tools/make_class_power.py

```diff
@@ -0,0 +1,206 @@
+        gparams['sigma8'] = config['Sigma8']
+    else:
+        #Pivot scale is by default 0.05 1/Mpc! This number is NOT what is reported by Planck.
+        gparams['A_s'] = config["PrimordialAmp"]
+    return gparams
+
+def make_class_power(paramfile, external_pk = None):
+    """Main routine: parses a parameter file and makes a matter power spectrum.
+    Will not over-write power spectra if already present.
+    Options are loaded from the MP-GenIC parameter file.
+    Supported:
+        - Omega_fld and DE parameters.
+        - Massive neutrinos.
+        - Using Sigma8 to set the power spectrum scale.
+        - Different transfer functions.
+    Not supported:
+        - Warm dark matter power spectra.
+        - Rescaling with different transfer functions."""
+    config = configobj.ConfigObj(infile=paramfile, configspec=GenICconfigspec, file_error=True)
+    #Input sanitisation
+    _check_genic_config(config)
+
+    #Load high precision defaults
+    pre_params = classylss.load_precision('pk_ref.pre')
+    gparams = _build_cosmology_params(config)
+    pre_params.update(gparams)
+    redshift = config['Redshift']
+    if config['InputPowerRedshift'] >= 0:
+        redshift = config['InputPowerRedshift']
+    #Pass options for the power spectrum
+    boxmpc = config['BoxSize'] / config['InputSpectrum_UnitLength_in_cm'] * config['UnitLength_in_cm']
+    maxk = 2*math.pi/boxmpc*config['Ngrid']*4
+    powerparams = {'output': 'dTk mPk', 'P_k_max_h/Mpc' : maxk, "z_pk": config['InputFutureRedshift'], "z_max_pk" : redshift}
+    pre_params.update(powerparams)
+
+    #Specify an external primordial power spectrum
+    if external_pk is not None:
+        pre_params['P_k_ini'] = "external_pk"
+        pre_params["command"] = "cat ",external_pk
+
+    #Make the power spectra module
+    engine = CLASS.ClassEngine(pre_params)
+    powspec = CLASS.Spectra(engine)
+    bg = CLASS.Background(engine)
+
+    #Save directory
+    sdir = os.path.split(paramfile)[0]
+    #Get and save the transfer functions if needed
+    if config['DifferentTransferFunctions'] == 1.:
+        trans = powspec.get_transfer(z=redshift)
+        tfile = os.path.join(sdir, config['FileWithTransferFunction'])
+        save_transfer(trans, tfile, bg, redshift)
+        transfut = powspec.get_transfer(z=config['InputFutureRedshift'])
+        tfile = os.path.join(sdir, config['FileWithFutureTransferFunction'])
+        save_transfer(transfut, tfile, bg, config['InputFutureRedshift'])
+    #Get and save the matter power spectrum
+    mink = 2*math.pi/boxmpc/100
+    kk = np.logspace(np.log10(mink), np.log10(maxk), max(config['Ngrid']*2,300))
+    pk_lin = powspec.get_pklin(k=kk, z=redshift)
+    pkfile = os.path.join(sdir, config['FileWithInputSpectrum'])
+    if os.path.exists(pkfile):
+        raise IOError("Refusing to write to existing file: ",pkfile)
+    np.savetxt(pkfile, np.vstack([kk, pk_lin]).T)
+
+def save_transfer(transfer, transferfile, bg, redshift):
```

Comment:
We shall probably add this functionality into classylss.

Reply:
Don't do that yet - I have a patch to class pending to change the transfer function output to better match CAMB/CMBFAST and if that gets merged this function will need to change.

Reply:
Sure.